### PR TITLE
Modify RenderViewToString sample to show how it could be used to process a view concurrently

### PR DESCRIPTION
**Pull request by pranavkm**

### CODE REVIEW

**Comment by pranavkm** [comment]
cc @ajaybhargavb

File: samples/Mvc.RenderViewToString/RazorViewToStringRenderer.cs [hunk 1]
```diff
@@ -30,7 +31,7 @@ public class RazorViewToStringRenderer
             _serviceProvider = serviceProvider;
         }
 
-        public string RenderViewToString<TModel>(string name, TModel model)
+        public async Task<string> RenderViewToString<TModel>(string name, TModel model)
```
Comment (ajaybhargavb) [inline samples/Mvc.RenderViewToString/RazorViewToStringRenderer.cs]: Should this be `RenderViewToStringAsync`?

File: samples/Mvc.RenderViewToString/Program.cs [hunk 2]
```diff
@@ -32,18 +35,26 @@ public static string RenderView(string customApplicationBasePath = null)
...
             services.AddSingleton<EmailReportGenerator>();
 
-            var provider = services.BuildServiceProvider();
-            var helper = provider.GetRequiredService<RazorViewToStringRenderer>();
+            var serviceProvider = services.BuildServiceProvider();
+            return serviceProvider.GetRequiredService<IServiceScopeFactory>();
+        }
 
-            var model = new EmailViewModel
+        public static Task<string> RenderViewAsync(IServiceScopeFactory scopeFactory)
```
Comment (ajaybhargavb) [inline samples/Mvc.RenderViewToString/Program.cs]: Can this name stay as `RenderView` and return `.Result`? The real async stuff happens in the helper right?

Comment (pranavkm) [inline samples/Mvc.RenderViewToString/Program.cs]: Presumably you want to do something like `Task.WaitAll(contents.Select(model => RenderViewAsync(serviceScopeFactory, model));`

File: samples/Mvc.RenderViewToString/RazorViewToStringRenderer.cs (hunk 1 above)
Comment (pranavkm) [inline samples/Mvc.RenderViewToString/RazorViewToStringRenderer.cs]: Sure

File: samples/Mvc.RenderViewToString/Program.cs (hunk 2 above)
Comment (ajaybhargavb) [inline samples/Mvc.RenderViewToString/Program.cs]: :+1:

**Comment by ajaybhargavb** [comment]
:shipit:

**Pull request closed without merging**